Lib/unittest/test/test_case.py

```diff
@@ -1777,6 +1777,31 @@ def test_something(self):
             self.assertEqual(len(result.skipped), 1)
             self.assertEqual(result.testsRun, 1)
 
+    def testSkippingDebugMode(self):
+        def _skip(self=None):
+            raise unittest.SkipTest('some reason')
+        def nothing(self):
+            pass
+
+        class Test1(unittest.TestCase):
+            test_something = _skip
+
+        class Test2(unittest.TestCase):
+            setUp = _skip
+            test_something = nothing
+
+        class Test3(unittest.TestCase):
+            test_something = nothing
+            tearDown = _skip
+
+        class Test4(unittest.TestCase):
+            def test_something(self):
+                self.addCleanup(_skip)
+
+        for klass in (Test1, Test2, Test3, Test4):
```

Comment:
What did you mean with "klass"?

Reply:
This is a common variable name used throughout this file (see line 1774 for example), it's a variable representing a class.